python/kastore/__init__.py

```diff
@@ -27,25 +28,28 @@ def _raise_unknown_engine():
     raise ValueError("unknown engine")
 
 
-def load(filename, read_all=False, key_encoding="utf-8", engine=PY_ENGINE):
+def load(file, read_all=False, key_encoding="utf-8", engine=PY_ENGINE):
     """
     Loads a store from the specified file.
 
+    :param str file: The path of the file to load, or a file-like object
+        with a ``read()`` method.
     :param bool read_all: If True, read the entire file into memory. This
         optimisation is useful when all the data will be needed, saving some
         malloc and fread overhead.
-    :param str filename: The path of the file to load.
     :param str key_encoding: The encoding to use when converting the keys from
         raw bytes.
     :param str engine: The underlying implementation to use.
     :return: A dict-like object mapping the key-array pairs.
     """
     if engine == PY_ENGINE:
-        return store.load(filename, read_all=read_all, key_encoding=key_encoding)
+        return store.load(file, read_all=read_all, key_encoding=key_encoding)
     elif engine == C_ENGINE:
         _check_low_level_module()
+        if isinstance(file, pathlib.PurePath):
```

Comment:
Does PyUnicode_EncodeFSDefault not handle a Pathlib instance?

Reply:
It would seem not. Passing in a pathlib.Path failed without this, unfortunately.

Reply:
Oh, and I forgot to note it elsewhere, but...

With current HEAD, the python API using PY_ENGINE and the C API should both be able to handle unicode filenames. But I'm pretty sure the python API using the C_ENGINE won't. This change should fix that. I never checked any unicode filenames though.